ci/ci/build.py

```diff
@@ -66,21 +66,32 @@ def __init__(self, code, scope, json, name_step):
 
 
 class BuildConfiguration:
-    def __init__(self, code, config_str, scope, profile=None):
+    def __init__(self, code, config_str, scope, requested_steps=None):
         config = yaml.safe_load(config_str)
         name_step = {}
         self.steps = []
 
-        if profile:
-            log.info(f"Constructing build configuration with following profile: {profile}")
+        if requested_steps:
+            log.info(f"Constructing build configuration with steps: {requested_steps}")
 
         for step_config in config['steps']:
             step_params = StepParameters(code, scope, step_config, name_step)
-
-            if profile is None or step_params.json['name'] in profile:
-                step = Step.from_json(step_params)
-                self.steps.append(step)
-                name_step[step.name] = step
+            step = Step.from_json(step_params)
+            self.steps.append(step)
+            name_step[step.name] = step
+
+        # transitively close requested_steps over dependenies
+        if requested_steps:
+            visited = set()
+            def request(step):
+                if step not in visited:
+                    visited.add(step)
+                    for s2 in step.deps:
+                        request(s2)
+            for s in requested_steps:
+                request(s)
+            self.steps = [s for s in self.steps if s in visited]
+            self.name_step = {s.name: s for s in self.steps}
```

Comment:
`name_step` doesn't have to be stored as an attribute of the`BuildConfiguration`, as it's not used anywhere else. It might be useful in the future, so if you want we can keep it, but then please make sure that it gets set even if `requested_steps = None`.

Reply:
Deleted, good catch.